DROD/MovementOrderHintEffect.cpp

```diff
@@ -0,0 +1,132 @@
+// $Id: MovementOrderHintEffect.h $
+
+/* ***** BEGIN LICENSE BLOCK *****
+ * Version: MPL 1.1
+ *
+ * The contents of this file are subject to the Mozilla Public License Version
+ * 1.1 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ * http://www.mozilla.org/MPL/
+ *
+ * Software distributed under the License is distributed on an "AS IS" basis,
+ * WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License
+ * for the specific language governing rights and limitations under the
+ * License.
+ *
+ * The Original Code is Deadly Rooms of Death.
+ *
+ * The Initial Developer of the Original Code is
+ * Caravel Software.
+ * Portions created by the Initial Developer are Copyright (C) 2025
+ * Caravel Software. All Rights Reserved.
+ *
+ * Contributor(s):
+ *
+ * ***** END LICENSE BLOCK ***** */
+
+#include "MovementOrderHintEffect.h"
+#include "RoomWidget.h"
+#include "DrodFontManager.h"
+#include "../DRODLib/CurrentGame.h"
+#include "../DRODLib/GameConstants.h"
+#include <BackEndLib/Assert.h>
+#include <FrontEndLib/Screen.h>
+
+//********************************************************************************
+CMovementOrderHintEffect::CMovementOrderHintEffect(
+	CWidget* pSetWidget, const CMonster* pMonster, int moveOrder)
+	: CEffect(pSetWidget, (UINT)-1, EMOVEORDERHINT)
+	, pMonster(const_cast<CMonster*>(pMonster))  //Though non-const, everything in this effect should leave monster state as-is
+	, wMoveOrder(moveOrder)
+	, pTextSurface(NULL)
+{
+	ASSERT(pSetWidget);
+	ASSERT(pSetWidget->GetType() == WT_Room);
+	ASSERT(pMonster);
+
+	this->pRoomWidget = DYN_CAST(CRoomWidget*, CWidget*, pSetWidget);
+	CDbRoom *pRoom = this->pRoomWidget->GetRoom();
+	ASSERT(pRoom);
+	if (pRoom) {
+		CCurrentGame* pGame = pRoom->GetCurrentGame();
+		ASSERT(pGame);
+		this->wValidTurn = pGame->wTurnNo;
+
+		//Prepare effect.
+		PrepWidget();
+	} else {
+		this->wValidTurn = 0;
+	}
+}
+
+//********************************************************************************
+CMovementOrderHintEffect::~CMovementOrderHintEffect()
+{
+	if (this->pTextSurface)
+		SDL_FreeSurface(this->pTextSurface);
+}
+
+//********************************************************************************
+bool CMovementOrderHintEffect::Update(const UINT wDeltaTime, const Uint32 dwTimeElapsed)
+{
+	const UINT wTurnNow = this->pRoomWidget->GetRoom()->GetCurrentGame()->wTurnNo;
```

Comment:
Suggest add:
ASSERT(this->pRoomWidget->GetRoom());
ASSERT(this->pRoomWidget->GetRoom()->GetCurrentGame());